### DISCUSSION TITLE
How to add a plugin for only a specific file type?

### DISCUSSION THREAD

**Discussion by erwin** [NvChad/NvChad#1586] [Q&A]
I searched through the Github Issues and read through the documentation, and though it's clear how to add plugins to the NvChad config, I don't see how to add a plugin for only a specific file type?

https://nvchad.com/config/plugins

Also, related, how add custom settings that only apply to a specific file type?

Both for plugins and for settings, I want my Markdown configuration to be much different than my regular coding oriented setups.

I saw discussion entries about adding plugins, but not about restricting them to certain file types:

https://github.com/NvChad/NvChad/discussions/1485

**Comment by siduck** [comment]
@erwin make an autocmd for it 

```lua
vim.api_nvim_create_autocmd("FileType", {
  pattern = "norg", -- suppose norg is the filetype name
  callback = function()
   -- do stuff 
  end,
})
```
